### PR TITLE
Add Cassandra supported versions and associated testing/documentation updates

### LINKED ISSUE #562: Test failure with Cassandra3InputFormatIT
```
$ docker run --name jg-cassandra -p 9160:9160 -p 9042:9042 -e CASSANDRA_START_RPC=true -d cassandra:3.10
$ mvn clean install -pl :janusgraph-hadoop-2 -DskipHBase -DskipCassandra -DskipCassandra3=false
...
Running org.janusgraph.hadoop.Cassandra3InputFormatIT
...
Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 804.834 sec <<< FAILURE! - in org.janusgraph.hadoop.Cassandra3InputFormatIT
testReadWideVertexWithManyProperties(org.janusgraph.hadoop.Cassandra3InputFormatIT)  Time elapsed: 306.026 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[0]> but was:<[44313]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.janusgraph.hadoop.AbstractInputFormatIT.testReadWideVertexWithManyProperties(AbstractInputFormatIT.java:83)
	...
```

### LINKED ISSUE #585: Update Cassandra supported versions
Cassandra should work with versions 2.2.z, 3.0.z, and 3.z. Recent Travis attempts by @sjudeng (https://travis-ci.org/sjudeng/janusgraph/builds/282098716) have shown a few errors

> The current build does still show a small number of CQL test failures. In 3.11.0 and 3.0.14 there are three failures in CQLStoreTest (testDefaultCFCompressor, testCustomCFCompressor and testDisableCFCompressor). In 2.2.10 there is one failure in CQLLogTest (testFuzzMessagesSerial). @pkendall64, @samant-orionhealth Have you observed these issues/failures before with newer C*, or any ideas how they might be resolved?

With 3.11.0 and 3.0.14 there are three failures in CQLStoreTest: testDefaultCFCompressor, testCustomCFCompressor and testDisableCFCompressor.
```
Tests run: 33, Failures: 3, Errors: 0, Skipped: 0, Time elapsed: 648.132 sec <<< FAILURE! - in org.janusgraph.diskstorage.cql.CQLStoreTest
testDefaultCFCompressor(org.janusgraph.diskstorage.cql.CQLStoreTest)  Time elapsed: 0.22 sec  <<< FAILURE!
java.lang.AssertionError: expected:<{sstable_compression=org.apache.cassandra.io.compress.LZ4Compressor, chunk_length_kb=64}> but was:<{chunk_length_in_kb=64, class=org.apache.cassandra.io.compress.LZ4Compressor}>
...

testCustomCFCompressor(org.janusgraph.diskstorage.cql.CQLStoreTest)  Time elapsed: 0.223 sec  <<< FAILURE!
java.lang.AssertionError: expected:<{sstable_compression=org.apache.cassandra.io.compress.DeflateCompressor, chunk_length_kb=128}> but was:<{chunk_length_in_kb=128, class=org.apache.cassandra.io.compress.DeflateCompressor}>
...

testDisableCFCompressor(org.janusgraph.diskstorage.cql.CQLStoreTest)  Time elapsed: 0.249 sec  <<< FAILURE!
java.lang.AssertionError: expected:<{}> but was:<{enabled=false}>
...
```

With 2.10.14 there is an error in CQLLogTest: testFuzzMessagesSerial.
```
Tests run: 11, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 80.187 sec <<< FAILURE! - in org.janusgraph.diskstorage.cql.CQLLogTest
testFuzzMessagesSerial(org.janusgraph.diskstorage.cql.CQLLogTest)  Time elapsed: 31.813 sec  <<< FAILURE!
java.lang.AssertionError: Did not read expected number of messages before timeout was reached (latch count is 32)
	at org.janusgraph.diskstorage.log.LogTest$LatchMessageReader.await(LogTest.java:325)
	at org.janusgraph.diskstorage.log.LogTest.testFuzzMessagesSerial(LogTest.java:212)
...
```

### PR DESCRIPTION
**Pull request by sjudeng**
Fixes #585. Also fixes #562 (sorry for the false alarm @twilmes, @nidmgh).

All default, Hadoop and TinkerPop tests pass for Cassandra 3.11.0, 3.0.14 and 2.2.10. See https://travis-ci.org/sjudeng/janusgraph/builds/284818802 (TinkerPop tests were run locally).

### CODE REVIEW

**Comment by nidmgh** [comment]
@sjudeng , many thanks. glad it is resolved.... Demai

File: TESTING.md [hunk 1]
```diff
@@ -56,48 +56,158 @@ mvn test -Dtest=BerkeleyJEGraphPerformanceMemoryTest
...
 ```
 
-### Running Thrift/CQL tests with ScyllaDB (requires Docker)
-
-To run Thrift or CQL tests with the [ScyllaDB](http://www.scylladb.com/) backend, ensure Docker is running and run tests under the `scylladb-test` Maven profile.
+Finally the `solr.test.version` property can be used to test against arbitrary Solr versions.
 
 ```bash
-mvn clean install -pl janusgraph-cql -Pscylladb-test
+mvn clean install -pl janusgraph-solr -Pdocker -Dsolr.test.version=5.5.4
```
[the remark this answers is not in the document]
Comment (sjudeng) [reply TESTING.md]: @amcp The ScyllaDB example was moved to the end of the document (see https://github.com/sjudeng/janusgraph/blob/cassandra-versions/TESTING.md#running-tests-with-scylladb).

**Comment by sjudeng** [comment]
@amcp To run tests across the versions I used Travis (https://travis-ci.org/sjudeng/janusgraph/builds/284818802) using the `.travis.yml.cassandra` file I included here as reference in case useful for future release testing. There are no changes to the actual project `.travis.yml` file.

I chose Travis over AWS for this because all jobs were within the 50 minute Travis timeout and it was able to complete the full build in under 3 hours, which is on the order of a standard JanusGraph Travis build. There is also additional test setup complexity (see https://github.com/sjudeng/janusgraph/blob/cassandra-versions/.travis.yml.cassandra#L52-L59) that I don't think the Codepipelines module is currently setup to support.

**Comment by pluradj** [comment]
Updating codepipelines should be a separate work item. I don't know if the LF would be able to provide us with an AWS account, but I wouldn't want to force contributors to pay for AWS resources.

File: janusgraph-hadoop-parent/janusgraph-hadoop-core/src/test/java/org/janusgraph/hadoop/AbstractInputFormatIT.java [hunk 2]
```diff
@@ -77,11 +78,10 @@ public void testReadWideVertexWithManyProperties() throws Exception {
...
         propertiesOnVertex = t.V().valueMap().next();
-        valuesOnP = (List)propertiesOnVertex.values().iterator().next();
-        assertEquals(numProps, valuesOnP.size());
-        for (int i = 0; i < numProps; i++) {
-            assertEquals(Integer.toString(i), valuesOnP.get(i).toString());
-        }
+        final Set<?> observedValuesOnP = ImmutableSet.copyOf((List)propertiesOnVertex.values().iterator().next());
+        assertEquals(numProps, observedValuesOnP.size());
+        // order may not be preserved in multi-value properties
+        assertEquals(0, Sets.symmetricDifference(observedValuesOnP, ImmutableSet.copyOf(valuesOnP)).size());
```
Comment (mbrukman) [inline janusgraph-hadoop-parent/janusgraph-hadoop-core/src/test/java/org/janusgraph/hadoop/AbstractInputFormatIT.java]: What about `assertTrue(Sets.symmetricDifference(...).isEmpty());` ?

Or `assertThat(observedValuesOnP).equals(ImmutableSet.copyOf(valuesOnP))` (or similar)? This appears to be testing for set equality; do we need a symmetric difference if we only care that it's 0?

File: janusgraph-cassandra/src/test/resources/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#
```
Comment (mbrukman) [inline janusgraph-cassandra/src/test/resources/docker-compose.yml]: Drop line?

**Comment by sjudeng** [comment]
@mbrukman Good catches, I'll push changes later today.

@amcp I could delete the `.travis.yml.cassandra` for now. Then the updates here would just be backend support (e.g. Docker scripts) for testing multiple Cassandra versions, without specifying whether testing is run locally, in Travis and/or AWS. We could then generalize the separate #584 to include AWS as well as Travis.

**Comment by mbrukman** [comment]
@sjudeng, @amcp — for expediency in moving forward with the 0.2.0 release, as well as effective usage of Travis (given our CPU budget for an open-source project), I would like to propose the following:

* [now] committing the `.travis.yml.cassandra` as-is to demonstrate how Cassandra compatibility was tested for this release, in case anyone wants proof or would like to recreate it themselves
* [soon] after the 0.2.0 release, consider refactoring the Travis CI configs so that there's a separate `cassandra` branch, similar to the `coverity_scan` branch which runs the full, extensive set of tests, but does not do all C* permutations for every PR (to save CPU cycles)
* [ongoing] this branch would be manually pushed, periodically (e.g., weekly, monthly, and definitely before every release) to ensure the testing happens
* [future] if that works out well, we can consider refactoring a few other version permutations such that we test the latest version and/or the most recent prior version for every PR, and test the full set a less often — this will help keep our Travis config from growing out of control with the blowup of many possible backend versions, which will only increase going forward

Thoughts?

**Comment by sjudeng** [comment]
@mbrukman Thanks, I think you just filled in the details on how to proceed on #584 after 0.2.0 release. I was thinking the cassandra test branch made sense as well. We will have to deal with merge conflicts anytime .travis.yml is touched on master, but they'll be easily handled (e.g. probably ignored).

**Comment by mbrukman** [comment]
@sjudeng — my hope is that the separate branches like `coverity_scan`, `cassandra`, etc. do not at all diverge from `master`, but just that some builds run only on specific branches, as was done for Coverity.

The way to do this would be to have no commits submitted to these separate branches directly, which means there are no merge conflicts, and every branch update is a `git pull master --fast-forward-only && git push` (or similar) update.

Would that work?

**Comment by amcp** [comment]
i agree to table the travis/pipelines changes for after release. ship it!

**Comment by amcp** [comment]
@sjudeng @mbrukman also agree that LF should have the option to use and not be required to use the pipelines package. So lets get that bit out of the way, and we can revisit with #584 .

File: janusgraph-cassandra/src/test/resources/docker-compose.yml (hunk 3 above)
Comment (sjudeng) [reply janusgraph-cassandra/src/test/resources/docker-compose.yml]: @mbrukman I did remove this line. Not sure why the comment isn't showing as outdated but I think I've seen that behavior before when the comment was on the first line in the file.

Comment (mbrukman) [reply janusgraph-cassandra/src/test/resources/docker-compose.yml]: Agreed, this is weird. The comment is misaligned given the new changes.

**Pull request merged**